### DISCUSSION TITLE
[CS0822] Allow function-local const-var for types which can be determined on compile-time

### DISCUSSION THREAD

**Discussion by vini-nu** [dotnet/csharplang#6940] [General]
If function local constant variable type can be determined in compile-time, then it should be allowed.

Currently it erroring with "CS0822: Implicitly-typed variables cannot be constant" syntax error.

This looks better readability:
![image](https://user-images.githubusercontent.com/6695244/216482427-d6c9a659-8db8-4185-97e2-5c2104e533db.png)
This looks worse readability:
![image](https://user-images.githubusercontent.com/6695244/216482529-9ee974db-f6f1-4c67-a25c-1cc948d09368.png)

```csharp
    private void Demo(CancellationToken ct = default)
    {
        const var splitOptions =
            StringSplitOptions.TrimEntries |
            StringSplitOptions.RemoveEmptyEntries;
    }
```

**Comment by TahirAhmadov** [comment]
This has been requested previously (#106 - thanks HaloFour). I think the current opinion is that the work is not worth the benefit, given other priorities. Personally, I would like `const splitOptions = StringSplitOptions.TrimEntries;` - not even `var`.

**Comment by HaloFour** [comment]
See: #106
